### PR TITLE
cert-inspection: add GatherCertsFromPlatformNamespaces, PKIRegistryInfo

### PR DESCRIPTION
**Pull request by vrutkovs**
This adds few functions used in e2e test (from https://github.com/openshift/origin/pull/28222)

### CODE REVIEW

File: pkg/certs/cert-inspection/certgraphapi/type_registry.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
+package certgraphapi
+
+import "fmt"
+
+type PKIRegistryInfo struct {
+	CertificateAuthorityBundles []PKIRegistryInClusterCABundle    `json:"certificateAuthorityBundles"`
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphapi/type_registry.go:6]: if you want these for in-cluster, name them in-cluster to let a different type tackle the on-disk ones.

File: pkg/certs/cert-inspection/certgraphapi/type_registry.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
+package certgraphapi
+
+import "fmt"
+
+type PKIRegistryInfo struct {
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphapi/type_registry.go:5]: given a godoc comment indicating whether or not you anticipate this growing fields to represent information like, "supports offline-cert-regeneration" or whether you see that as distinct.

File: pkg/certs/cert-inspection/certgraphapi/type_registry.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+type PKIRegistryInClusterCertKeyPair struct {
+	SecretLocation InClusterSecretLocation `json:"secretLocation"`
+
+	CertKeyInfo PKIRegistryCertKeyPairInfo `json:"certKeyInfo"`
+}
+
+type PKIRegistryCertKeyPairInfo struct {
+	OwningJiraComponent string `json:"owningJiraComponent"`
+	Description         string `json:"description"`
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphapi/type_registry.go:18]: indicate how long you're expecting this to be?  One sentence or one paragraph?

File: pkg/certs/cert-inspection/certgraphapi/type_registry.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+type PKIRegistryCertificateAuthorityInfo struct {
+	OwningJiraComponent string `json:"owningJiraComponent"`
+	Description         string `json:"description"`
+
+	//CertificateData []PKIRegistryCertKeyMetadata
+}
+
+type RegistrySecretByNamespaceName []PKIRegistryInClusterCertKeyPair
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphapi/type_registry.go:36]: move functions to a helper package to better match "normal" APIs.  We keep them separate to handle dependency chains better.

File: pkg/certs/cert-inspection/certgraphapi/type_registry.go (hunk 1 above)
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphapi/type_registry.go:6]: mock up an SSA tag here. I think it would be map

**Comment by deads2k** [comment]
minor comments, lgtm otherwise.

File: pkg/certs/cert-inspection/certgraphapi/type_registry.go (hunk 1 above)
Comment (vrutkovs) [reply pkg/certs/cert-inspection/certgraphapi/type_registry.go:6]: Lets rename `PKIRegistryInfo` into `ClusterPKIRegistryInfo` to differentiate it from on-disk certificates? We probably don't need to have different sets of on-disk registries for different topologies and platforms

File: pkg/certs/cert-inspection/certgraphanalysis/nameguessing.go [hunk 5]
```diff
@@ -55,22 +56,59 @@ var secretLocationToLogicalName = map[certgraphapi.InClusterSecretLocation]logic
 	newSecretLocation("", ""): newMeaning("", ""),
 }
 
-func guessMeaningForCertKeyPair(in certgraphapi.CertKeyPair) logicalMeaning {
-	for _, loc := range in.Spec.SecretLocations {
-		if meaning, ok := secretLocationToLogicalName[loc]; ok {
-			return meaning
-		}
+func formatSecretLocation(loc certgraphapi.InClusterSecretLocation, nodes map[string]int) certgraphapi.InClusterSecretLocation {
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphanalysis/nameguessing.go:59]: I'm ok with this approach as your starting point (won't block merge), but I suspect you'll actually want to scan everything and replace the nodename in all secret names, all namespaces, all dns names, all subjects, etc.

File: pkg/certs/cert-inspection/certgraphapi/type_registry.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
+package certgraphapi
+
+// PKIRegistryInfo holds information about TLS artifacts stored in etcd. This includes object location and metadata based on object annotations
+type PKIRegistryInfo struct {
+	// +mapType:=atomic
+	CertificateAuthorityBundles []PKIRegistryInClusterCABundle `json:"certificateAuthorityBundles"`
```
Comment (deads2k) [inline pkg/certs/cert-inspection/certgraphapi/type_registry.go:6]: I'll allow a followup, but I suspect this object grows into `inClusterCertificateAuthorityBundles` on `onDiskCertificateAuthorityBundles`

Comment (vrutkovs) [reply pkg/certs/cert-inspection/certgraphapi/type_registry.go:6]: I'd rather have `PKIRegistryInfo` include `OnDiskPKIRegistryInfo` and `ClusterPKIRegistryInfo` structs, but yeah, we could merge them in one structure later

**Pull request merged**